File: src/knowflow/api/v1/endpoints/knowledge.py
"""知识检索端点 - POST /knowledge/search 调 HybridRetriever.

只读检索, 不写库. 返回 top_k 检索结果(含 chunk 内容/分数/来源).
"""

from fastapi import APIRouter

from knowflow.api.deps import RetrieverDep
from knowflow.schemas.common import ApiResponse
from knowflow.schemas.knowledge import (
    ChunkResult,
    SearchRequest,
    SearchResponse,
)

router = APIRouter(prefix="/knowledge", tags=["knowledge"])


@router.post("/search", response_model=ApiResponse[SearchResponse])
async def search(
    req: SearchRequest,
    retriever: RetrieverDep,
) -> ApiResponse[SearchResponse]:
    """知识检索: hybrid → rerank → 缓存."""
    result = await retriever.retrieve(
        req.query,
        top_k=req.top_k,
        with_rerank=req.with_rerank,
    )
    chunks = [
        ChunkResult(
            chunk_id=c.chunk_id,
            content=c.content,
            score=c.score,
            source=c.source,
            doc_id=c.doc_id,
            doc_title=c.doc_title,
        )
        for c in result.chunks
    ]
    return ApiResponse(
        data=SearchResponse(
            query=result.query,
            chunks=chunks,
            latency_ms=result.latency_ms,
            cache_hit=result.cache_hit,
            total=len(chunks),
        )
    )

File: web/src/api/knowledge.ts
import { request } from './client';
import type { ApiResponse, GraphData, SearchResponse } from '@/types';

export const knowledgeApi = {
  /** 知识库检索（GraphRAG 全链路：Hybrid 召回 → 一跳扩展 → reranker 精排） */
  search: (query: string, topK = 5) =>
    request<SearchResponse>('/knowledge/search', {
      method: 'POST',
      body: { query, top_k: topK },
    }),

  /** 知识图谱数据：实体节点 + 关系边（doc_id 为空取全库） */
  getGraph: async (docId?: number, limit = 200) => {
    const r = await request<ApiResponse<GraphData>>('/knowledge/graph', {
      query: { doc_id: docId, limit },
    });
    return r.data;
  },
};
